### ISSUE TITLE
[1.0] Oscillator-Controlled lights broken past ~2.6 Hz.

### ISSUE THREAD

**Issue by SeaverBeaver** [Regalis11/Barotrauma#11243]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

When using square-wave oscillators to control the state of lights, whether they be light components or "lighting" lights, a "ramp up" delay is introduced when the light is oscillated on, meaning that when the oscillator component is set higher than 2.6Hz, the light becomes barely visible. Any higher than 2.8Hz and it never turns on in the first place.

https://user-images.githubusercontent.com/95244788/218673996-4534a583-52ef-4883-b859-b28a0d0a7cdc.mp4

In the top right corner, just above the cursor, you'll find that the "FIRE" warning light, set at 2.6Hz, is the worst offender, but every other oscillator controlled light in the room is doing the same.

It should be noted that the lights in question do not draw power, and in the current live version oscillate on/off instantly.

### Reproduction steps

1. Open the sub editor in the 1.0 build.
2. Place a square-wave oscillator component set to 3Hz, connected to a light.
3. Place an identical light set to blink at 3Hz.
4. Observe the bug with the oscillator connected light.

### Bug prevalence

Happens every time I play

### Version

Faction/endgame test branch

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Addressed in https://github.com/Regalis11/Barotrauma-development/commit/36eac378e0efada9e4ee194c6cea8cd626c24990

**Comment by NilanthAnimosus** [comment]
Tested against Fix/light-fix-part-3 commit https://github.com/Regalis11/Barotrauma-development/commit/8a74178e6280275e7590eafd56a8792d9aab08b0, no issues found, ready for merging

Notes:

![image](https://user-images.githubusercontent.com/29177976/219051618-2a07a91d-287f-4a89-960e-ccf7a9d1b708.png)

Tested using a series of different oscillator frequencies set to square mode, both colour and on the lights on/off with no issues.

Additionally the lights still correctly light up the specific areas when on as they should (Switches, lamps etc) with their light emitting textures. and the AI Targets generated by items like the flashlight only appear when its powered and held, worn (IE. head slot) or dropped (but not in inventory with any means I tried).
